src/tlo/methods/wasting.py

```diff
@@ -1,41 +1,1442 @@
+        item_code2 = pd.unique(consumables.loc[consumables['Items'] == 'SAM medicines', 'Item_Code'])[0]
+
+        # # check availability of consumables
+        if self.get_consumables(item_code1) and self.get_consumables(item_code2):
+            logger.debug(key='debug', data='consumables available, so use it.')
+            # Log that the treatment is provided:
+            df.at[person_id, 'un_acute_malnutrition_tx_start_date'] = self.sim.date
+            df.at[person_id, 'un_am_discharge_date'] = self.sim.date + DateOffset(weeks=4)
+            df.at[person_id, 'un_am_treatment_type'] = 'inpatient_care'
+            self.module.do_when_am_treatment(person_id, intervention='ITC')
+        else:
+            logger.debug(key='debug', data="consumables not available, so can't use it.")
+
+    def did_not_run(self):
+        logger.debug("HSI_inpatient_care_for_complicated_SAM: did not run")
         pass
 
-    def on_birth(self, mother, child):
+
+class WastingModels:
+    """ houses all wasting linear models """
+
+    def __init__(self, module):
+        self.module = module
+        self.params = module.parameters
+        self.rng = module.rng
+
+        # linear model to predict the incidence of wasting
+        self.__Wasting_Incidence = LinearModel.multiplicative(
+            Predictor('age_exact_years').when('<0.5', self.params['base_inc_rate_wasting_by_agegp'][0])
+            .when('<1.0', self.params['base_inc_rate_wasting_by_agegp'][1])
+            .when('.between(1,1.9999)', self.params['base_inc_rate_wasting_by_agegp'][2])
+            .when('.between(2,2.9999)', self.params['base_inc_rate_wasting_by_agegp'][3])
+            .when('.between(3,3.9999)', self.params['base_inc_rate_wasting_by_agegp'][4])
+            .when('.between(4,4.9999)', self.params['base_inc_rate_wasting_by_agegp'][5])
+            .otherwise(0.0),
+            Predictor().when('(nb_size_for_gestational_age == "small_for_gestational_age") &'
+                             '(nb_late_preterm == False) & (nb_early_preterm == False)',
+                             self.params['rr_wasting_SGA_and_term']),
+            Predictor().when('(nb_size_for_gestational_age == "small_for_gestational_age") & '
+                             '(nb_late_preterm == True) | (nb_early_preterm == True)',
+                             self.params['rr_wasting_SGA_and_preterm']),
+            Predictor().when('(nb_size_for_gestational_age == "average_for_gestational_age") &'
+                             '(nb_late_preterm == True) | (nb_early_preterm == True)',
+                             self.params['rr_wasting_preterm_and_AGA']),
+            Predictor('li_wealth').apply(lambda x: 1 if x == 1 else (x - 1) ** (
+                self.params['rr_wasting_wealth_level'])),
+        )
+
+        # a linear model to predict the probability of individual's recovery from moderate acute malnutrition
+        self.__Acute_Malnutrition_Recovery_MAM = LinearModel.multiplicative(
```

Comment:
(same as above) see the wiki on LinearModels for maximising efficiency, using exhaustive and mutually exclusive options.